src/pu-emmc.c

```diff
@@ -512,6 +514,42 @@ pu_emmc_init(G_GNUC_UNUSED PuEmmc *self)
 {
 }
 
+static PedSector
+pu_emmc_get_partition_table_size(PuEmmc *emmc)
+{
+    if (g_str_equal(emmc->disktype->name, "msdos"))
+        return PARTITION_TABLE_SIZE_MSDOS;
+    else if (g_str_equal(emmc->disktype->name, "gpt"))
+        return PARTITION_TABLE_SIZE_GPT;
+    return 1;
+}
+
+static gboolean
+pu_emmc_get_input_size(PuEmmcInput *input,
+                       const gchar *prefix,
+                       GError **error)
```

Comment:
I would expect for a getter function to return the requested value (the input's size). I think it is better to pass any file path as a string to this function, instead of `input` and `prefix`, and return the file's size. In case of error (e.g. file does not exist), returning `0` seems adequate. GError can still be used then.